.github/workflows/ci.yml

```diff
@@ -87,6 +87,48 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true' && matrix.cache == 'true'
         run: rm -rf ~/.m2/repository/io/trino/trino-*
 
+  reproducible-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
+          ref: |
+            ${{ github.event_name == 'repository_dispatch' &&
+                github.event.client_payload.pull_request.head.sha == github.event.client_payload.slash_command.args.named.sha &&
+                format('refs/pull/{0}/head', github.event.client_payload.pull_request.number) || '' }}
+      - uses: ./.github/actions/setup
+        with:
+          cache: 'false'
+          cleanup-node: true
+      - name: First build to a local repository
+        # All dependencies are downloaded to and artifacts generated to build_1 directory, .m2 isn't used
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} -Dmaven.repo.local=build_1
+      - name: Second build to a local repository
+        # All dependencies are downloaded to and artifacts generated to build_2 directory, .m2 isn't used
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} -Dmaven.repo.local=build_2
+      - name: Compare builds
+        # Excluding maven metadata files - downloaded dependencies and generated artifacts should be exactly the same
+        run: |
+          set -e
```

Comment:
isn't this the default?

Reply:
I have no idea

Reply:
can you check? it is important that it is

(BTW it **should** be thanks to https://github.com/trinodb/trino/blob/e54af31685cbd4f33b34ae35df76793c724623ab/.github/workflows/ci.yml#L15 but i don't know whether this works)